pkg/stream/retry_io.go

```diff
@@ -0,0 +1,100 @@
+package stream
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/justtrackio/gosoline/pkg/cfg"
+	"github.com/justtrackio/gosoline/pkg/cloud/aws/sqs"
+	"github.com/justtrackio/gosoline/pkg/log"
+)
+
+type RetryHandlerIO struct {
+	input    AcknowledgeableInput
+	output   Output
+	settings *RetrySettings
+}
+
+func NewRetryHandlerIO(ctx context.Context, config cfg.Config, logger log.Logger, settings *RetrySettings) (RetryHandler, error) {
+	var err error
+	var input AcknowledgeableInput
+	var output Output
+
+	inputSettings := &SqsInputSettings{
+		AppId:               settings.AppId,
+		QueueId:             settings.QueueName,
+		MaxNumberOfMessages: 10,
+		WaitTime:            10,
+		VisibilityTimeout:   int(settings.After.Seconds()),
+		RunnerCount:         1,
+		RedrivePolicy: sqs.RedrivePolicy{
+			Enabled:         true,
+			MaxReceiveCount: settings.MaxAttempts,
+		},
+		ClientName:   "",
+		Unmarshaller: UnmarshallerMsg,
+	}
+
+	if input, err = NewSqsInput(ctx, config, logger, inputSettings); err != nil {
+		return nil, fmt.Errorf("can not create input: %w", err)
+	}
+
+	outputSettings := &SqsOutputSettings{
+		AppId:             settings.AppId,
```

Comment:
I think I would initialize all fields here from the inputSettings, makes it clearer that the two are related

Reply:
done